app/views/GermlineView/components/Report/index.tsx

```diff
@@ -0,0 +1,190 @@
+import snackbar from '@/services/SnackbarUtils';
+import StrikethroughCell from './components/StrikethroughCell';
+import EditDialog from './components/EditDialog';
+import Reviews from './components/Reviews';
+import columnDefs from './columnDefs';
+import { GermlineReportType } from '../../types';
+
+import './index.scss';
+
+const GermlineReport = (): JSX.Element => {
+  const { ident } = useParams();
+  const { colApi, onGridReady } = useGrid();
+  const history = useHistory();
+  const [report, setReport] = useState<GermlineReportType>();
+
+  const [isLoading, setIsLoading] = useState(true);
+  const [showAllColumns, setShowAllColumns] = useState(false);
+  const [showEditDialog, setShowEditDialog] = useState(false);
+  const [showAlertDialog, setShowAlertDialog] = useState(false);
+  const [editData, setEditData] = useState();
+
+  useEffect(() => {
+    if (ident) {
+      const getData = async () => {
+        const reportResp = await api.get(
+          `/germline-small-mutation-reports/${ident}`,
+          {},
+        ).request();
+        setReport(reportResp);
+        setIsLoading(false);
+      };
+      getData();
+    }
+  }, [ident, setReport]);
+
+  useEffect(() => {
+    if (colApi) {
+      const colIds = colApi.getAllColumns().filter((col) => !col.colDef.autoHeight).map((col) => col.colId);
+      colApi.autoSizeColumns(colIds, false);
+    }
+  }, [colApi]);
+
+  const handleShowAllColumns = useCallback((event: React.ChangeEvent<HTMLInputElement>) => {
+    const { checked } = event.target;
+    setShowAllColumns(checked);
+
+    if (checked) {
+      colApi.applyColumnState({ defaultState: { hide: false } });
+    } else {
+      colApi.resetColumnState();
+    }
+  }, [colApi]);
+
+  const onEdit = useCallback((rowData) => {
+    setShowEditDialog(true);
+    setEditData(rowData);
+  }, []);
+
+  const handleEditClose = useCallback((newRow) => {
+    if (newRow) {
+      const newVariants = [...report.variants];
+      const index = newVariants.findIndex((variant) => variant.ident === newRow.ident);
+      newVariants[index] = newRow;
+      setReport((prevVal) => ({ ...prevVal, variants: newVariants }));
+    }
+    setShowEditDialog(false);
+  }, [report]);
+
+  const RowActionCellRenderer = useCallback((row) => (
+    <ActionCellRenderer
+      onEdit={() => onEdit(row.node.data)}
+      // eslint-disable-next-line react/jsx-props-no-spreading
+      {...row}
+    />
+  ), [onEdit]);
+
+  const handleConfirmDelete = useCallback(async (confirm) => {
+    setShowAlertDialog(false);
+    if (confirm) {
+      try {
+        await api.del(`/germline-small-mutation-reports/${report.ident}`, {}, {}).request();
+        snackbar.success('Report deleted');
+        history.push('/germline');
+      } catch (err) {
+        snackbar.error(`Error deleting report: ${err}`);
+      }
+    }
+  }, [report, history]);
+
+  return (
+    <ReportContext.Provider value={{ report, setReport }}>
```

Comment:
it's probably worth it to make a `GermlineReportContext` and `useGermlineReport` custom hook instead of using `ReportContext` here directly, so all the components that I've seen flagged because TS can't tell if report in this provider is `Report` or `GermlineReport`